src/ng/rootScope.js

```diff
@@ -296,7 +296,7 @@ function $RootScopeProvider() {
        * - When `objectEquality == true`, inequality of the `watchExpression` is determined
        *   according to the {@link angular.equals} function. To save the value of the object for
        *   later comparison, the {@link angular.copy} function is used. This therefore means that
-       *   watching complex objects will have adverse memory and performance implications.
+       *   watching complex objects will have adverse memory and performance implications. This should not be used to watch for objects that are or contain `File` objects.
```

Comment:
I think this should go into its own bullet point, with an explanation why it's not recommended.
Or if it is not recommended because file objects use a lot of memory, then this should also be pointed out.